packages/core-data/src/actions.js

```diff
@@ -357,28 +357,34 @@ export function* saveEntityRecord(
 				}
 			}
 
-			// We perform an optimistic update here to clear all the edits that
-			// will be persisted so that if the server filters them, the new
-			// filtered values are always accepted.
-			persistedEntity = yield select(
-				'getEntityRecord',
-				kind,
-				name,
-				recordId
-			);
-			currentEdits = yield select(
-				'getEntityRecordEdits',
-				kind,
-				name,
-				recordId
-			);
-			yield receiveEntityRecords( kind, name, { ...persistedEntity, ...data }, undefined, true );
+			/*
+				// the code below issues an API fetch action against the server and gets the outdates pre-modification state of the record
+				// and since getEntityRecord calls receiveEntityRecords, this outdated state is rendered to the user, causing an unstable (flickering)
+				// intermediate UI state
+
+				//////// disabled //////////
+					// We perform an optimistic update here to clear all the edits that
+					// will be persisted so that if the server filters them, the new
+					// filtered values are always accepted.
+					persistedEntity = yield select(
+						'getEntityRecord',
+						kind,
+						name,
+						recordId
+					);
+				//////// disabled //////////
+			*/
+
+			// optimistic update subscribers (eg trigger re-renders) with the modifications applied to the record
+			// and on their way to be persisted on the server
+			yield receiveEntityRecords( kind, name, { ...data }, undefined, true );
```

Comment:
I know we had a lot of back and forth here and this is a very sensitive code to change (as you can see from the e2e tests). cc @epiqueras

Reply:
Ha, I knew I'll break things my this change.

I must have written this in the PR comment. I don't think my patch solves the problem well. But I think it does pin it down; the current version is publishing the outdated state when it shouldn't; this is a bug and it's shows on the UI whenever you update the state.

Would you like me to setup a demo repo?

Reply:
It's breaking because you are optimistically updating the record to become the subset of edited properties. We need the `persistedRecord` spread at the beginning of that object. `receiveEntityRecords` sets the object, it doesn't merge properties to what's already in the store.

We also need `currentEdits` for later logic, so we can't remove that code. `getEntityRecordEdits` shouldn't be a problem here.

The issue you are experiencing is happening because `getEntityRecord` triggers a resolver the first time it fires for a given set of parameters. This resolver fetches and sets the record. You usually don't call `saveEntityRecord` before having called `getEntityRecord`, but in this case, you are probably calling `getEntityRecords` and then calling `saveEntityRecord` directly.

I think this is something we should fix here if anything because of the performance drag of that unnecessary extra request.

To solve this, we need to change that instance of `getEntityRecord` to one that does not trigger its resolver. I.e. an alias with a different name that calls it in its implementation, `getEntityRecordNoResolver(...) => getEntityRecord(...)` or something like that.

Reply:
We could also look at exposing an API for calling a selector without triggering its resolvers.

Reply:
Makes perfect sense. Can I introduce this selector with its docs and such?

Reply:
That would be great, yes. It's fine to do it in this PR if you add an update to the description. It would be great to add a unit test for it as well; "saveEntityRecord() does not trigger the `getEntityRecord` resolver."

Thanks for catching this!

Reply:
Done. But instead of testing that `saveEntityRecord` _doesn't_ issue an API request (kinda like proving a negative) I tested that it selects `getEntityRecordNoResolver` instead of `getEntityRecord`. 

Testing that it doesn't issue an API request will also be confusing to someone not fimilar with the history in my opinion. I'm open to changing it though.

All update the PR's message in the next couple hours.

Reply:
To be entirely correct, we should probably have an integration test for the store where we use `hasStartedResolution` to verify the resolver is not called or even better, an E2E test to reproduce the exact issue you experienced.

But we don't have that setup, and it would be kind of out of scope for this PR.

Proving a negative is fine if the intent is correct. For example, without proving a negative here, if in the future the registry API is changed so that nested selector calls also trigger resolvers, this would break silently.